### ISSUE TITLE
Document the way the require-dev section is treated

### ISSUE THREAD

**Issue by holtkamp** [Ocramius/PackageVersions#22]
It might be useful to document the way this plugin works. Its functioning seems obvious, but I was kind of surprised I had access to dependencies that were defined in ```require-dev``` sections on a production environment, where these dependencies are not physically installed.

[This feature](https://github.com/Ocramius/PackageVersions/blob/7519e4dca762484ac3ca507bfd133f9fe417c896/src/PackageVersions/Installer.php#L169-L171) enables us to load assets from a CDN using the exact same version as used when installed on a development environment. 

**Rationale**
To reduce the amount of installed assets, bower assets such as jQuery are in ```require-dev```. Apparently on a production environment the version can still be derived, allowing us to refer to the same version of jQuery on a CDN as when used in development. 😆

**Comment by Ocramius** [comment]
The problem being solved by this package is usually around codegen. Specifically, say you generate assets with a dependency set `X+Y+Z`, you want to re-generate (or resolve) cache paths depending on those generated paths against `X+Y+Z`, regardless if those are in dev dependencies or not.

The point is being able to re-generate paths and hashes depending on installed dependencies, disallowing reuse of hashes/paths/etc when an upgrade is performed.

Not sure how to convey that in a simpler wording through the `README.md`...

**Comment by holtkamp** [comment]
possibly I am just nitpicking, suggestion here: https://github.com/Ocramius/PackageVersions/compare/master...holtkamp:patch-1

**Comment by Ocramius** [comment]
@holtkamp that works, I just added some remarks, as I think it makes the sentence a bit hard to follow. The package doesn't contain much, so I'd rather keep it simple for the reader and for self-advertisement reasons.

**Comment by Ocramius** [comment]
Closing here, moved to #23
